pkg/tfbridge/detailed_diff.go

```diff
@@ -264,13 +274,48 @@ func (differ detailedDiffer) makePlainPropDiff(
 	return nil
 }
 
+// makeShortCircuitDiff is used for properties that are nil or computed in either the old or new state.
+// It makes sure to check recursively if the property will trigger a replacement.
+func (differ detailedDiffer) makeShortCircuitDiff(
+	path propertyPath, old, new resource.PropertyValue,
+) map[detailedDiffKey]*pulumirpc.PropertyDiff {
+	contract.Assertf(old.IsNull() || new.IsNull() || new.IsComputed(),
+		"short-circuit diff should only be used for nil properties")
+	if old.IsNull() && new.IsNull() {
+		return nil
+	}
+
+	baseDiff := makeBaseDiff(old, new)
+	contract.Assertf(baseDiff != undecidedDiff, "short-circuit diff could not determine diff kind")
+
+	propDiff := baseDiff.ToPropertyDiff()
+	if new.IsComputed() && propertyPathTriggersReplacement(path, differ.tfs, differ.ps) {
+		propDiff = promoteToReplace(propDiff)
+	} else if !new.IsNull() && !new.IsComputed() && propertyValueTriggersReplacement(path, new, differ.tfs, differ.ps) {
```

Comment:
Had to think about this one. This is suspect. I'm not sure this is right, not sure how to verify this. This analyzes schemas for all sub-paths realized in `new` property Value, without any consideration as to whether these sub-paths result in diff or no-diff compared to the `old` value. Why is that? Ah that's because this assumes old.IsNull.

Reply:
Do you think I should add a comment here? Which part is suspect?

Reply:
I grokked it but it required effort.